core/src/main/java/com/redhat/lightblue/client/response/LightblueHealthResponse.java

```diff
@@ -0,0 +1,83 @@
+package com.redhat.lightblue.client.response;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonToken;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.redhat.lightblue.client.LightblueException;
+
+/**
+ * @author ssaurabh
+ *
+ */
+/**
+ * @author ssaurabh
+ *
+ */
+public class LightblueHealthResponse extends AbstractLightblueResponse implements LightblueResponse {
+
+    private static final long serialVersionUID = 5383951657430651771L;
+
+    private boolean isHealthy;
+
+    public LightblueHealthResponse(JsonNode responseNode, Map<String, List<String>> headers)
+            throws LightblueResponseException, LightblueException {
+        super(responseNode, headers);
+    }
+
+    public LightblueHealthResponse(JsonNode responseNode, ObjectMapper mapper)
+            throws LightblueResponseException, LightblueException {
+        super(responseNode, null, mapper);
+    }
+
+    public LightblueHealthResponse(JsonNode responseNode, Map<String, List<String>> headers, ObjectMapper mapper)
+            throws LightblueResponseException, LightblueException {
+        super(responseNode, headers, mapper);
+    }
+
+    public LightblueHealthResponse(String responseText)
+            throws LightblueParseException, LightblueResponseException, LightblueException {
+        super(responseText, null);
+    }
+
+    public LightblueHealthResponse(String responseText, Map<String, List<String>> headers)
+            throws LightblueParseException, LightblueResponseException, LightblueException {
+        super(responseText, headers);
+    }
+
+    public LightblueHealthResponse(String responseText, Map<String, List<String>> headers, ObjectMapper mapper)
+            throws LightblueParseException, LightblueResponseException, LightblueException {
+        super(responseText, headers, mapper);
+    }
+
+    public boolean isHealthy() {
+        return parseHealthy();
+    }
+
+    
+    /**
+     * parses the lightblue health response and returns true only if healthy is
+     * true for all the nodes in the response, otherwise returns false
```

Comment:
Going back to the /health vs /diagnostics distinction, this looks like you are talking about the /diagnostics endpoint (which makes sense, because this is the only one that exists right now).

For /diagnostics, we should be careful encoding logic in the client that says what "healthy" means. It depends on use case and lightblue server configuration.  I would probably consider not providing this kind of API in the client, and instead have a class that encapsulates the response and allows the application using the client to easily examine health of controllers, etc, that may be included in the diagnostics response. I forget what the schema of the response is but basically it should follow that.